sql/modules/Reconciliation.sql

```diff
@@ -681,9 +689,11 @@ account.  For asset and expense accounts this is the debit balance, for others
 this is the credit balance.$$;
 
 CREATE OR REPLACE VIEW recon_payee AS
- SELECT n.name AS payee, rr.id, rr.report_id, rr.scn, rr.their_balance, rr.our_balance, rr.errorcode, rr."user", rr.clear_time, rr.insert_time, rr.trans_type, rr.post_date, rr.ledger_id, ac.voucher_id, rr.overlook, rr.cleared
+ SELECT DISTINCT ON (rr.id)
```

Comment:
Do you have better implementation?

Reply:
I'm not sure. In order to progress with the rest of it: is it required to have this fix to correct the functionality you need fixed? that is: we're trying to fix the Source being filled in the reconciliation report. Is this part of that, or is it an unrelated find that you're correcting along the way. (My analysis seems to indicate it's the latter.)

Note that I'm not saying this in order not to have to fix it, but there's the matter of prioritization too.

Reply:
Yes, it is unrelated with source. I found that the `Description` column in the recon report is fill with `payee` but it is not correctly link because of the changes in `payments` and `payment_link` table. That's why I fixed it along the way.

Reply:
Ok. I think this query *and* fix are too complex. The reason for the original query to join in all the other components was solely to retrieve the `voucher_id`. Now, looking at the use of the view, it's being used in two places: `reconciliation__report_details_payee` and `reconciliation__report_details_payee_with_days`.

The first of the two returns all columns and is invoked in a single place: `old/lib/LedgerSMB/DBObject/Reconciliation.pm`, in the `get_report_lines()` function. That function isn't called *anywhere* in `old/` or `lib/`. Which means as much to me as that this code is dead code... (No, I'm not expecting you to do that analysis all by yourself! :-) ) I'll create a PR to remove it to see whether we can do away with this.